File: src/Tests/Nest.Tests/Search APIs/Request Body Search/From and Size.fs
namespace ``Search APIs``

open Xunit
open Nest
open System
open Nest.Tests
open FsUnit

type Rec = { x: int }

module ``Request Body Search`` =
    type ``From and Size``() =
        inherit BaseTest<ISearchRequest, SearchDescriptor<Rec>, SearchRequest>()

(**
Pagination of results can be done by using the from and size parameters. 
The from parameter defines the offset from the first result you want to fetch. 
The size parameter allows you to configure the maximum amount of hits to be returned.
*)
        override test.ExpectedJson(map) = 
            map
                .Add("from", 12)
                .Add("size", 10)

        override test.Initializer = new SearchRequest (From= Nullable 12, Size= Nullable 10);

        override test.Fluent(descriptor) = 
            descriptor 
                .From(12)
                .Size(10)
        
        [<Fact>]
        member test.``Search behaves properly``() = 
            let result = test.Client.Search<Rec>(test.Fluent)
            result.IsValid |> should be True

File: src/Tests/Search/Request/FromAndSize.cs
using System;
using Nest;
using Tests.Framework.Integration;
using Tests.Framework.MockData;
using Tests.SearchAPIs;

namespace Tests.Search.Request
{
	public class FromAndSize
	{
		/**
		 * Pagination 2 of results can be done by using the from and size parameters. 
		 * The from parameter defines the offset from the first result you want to fetch. 
		 * The size parameter allows you to configure the maximum amount of hits to be returned.
		 */

		public class Usage : SearchUsageBase
		{
			public Usage(ReadOnlyIntegration i) : base(i) {} 

			protected override object ExpectedJson =>
				new { from = 10, size = 12 };

			public override int ExpectStatusCode => 200;

			public override bool ExpectIsValid => true;

			public override string UrlPath => "/project/project/_search";

			protected override SearchRequest<Project> Initializer =>
				new SearchRequest<Project>()
				{
					From = 10,
					Size = 12
				};

			protected override Func<SearchDescriptor<Project>, ISearchRequest> Fluent => s => s
					.From(10)
					.Size(12);
		}
	}
}
